### DISCUSSION TITLE
Setting value for integer dict key

### DISCUSSION THREAD

**Discussion by paulmelis** [wjakob/nanobind#29] [Q&A] [answered]
I must be stupid, but I don't see how I can set an integer key on an `nb::dict`. Something like `nb::dict res; res[1] = "a";` gives a runtime error `TypeError: dict is not a sequence`, apparently interpreting it as a list. Yet, `res["1"] = "a";` works. Is there a specific call I need to use?

**Comment by wjakob** [comment, marked as answer]
Integer indexing is internally wired up to the sequence protocol, which dictionaries do not support. As of commit https://github.com/wjakob/nanobind/commit/93a653c4cf12558fac9330423a6cfed3d17bbd87, there is a workaround that you can use:

```cpp
res[nb::int_(value)] = "a";
```
